### PR TITLE
Expandable console support

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

**Review by jacob314 — COMMENTED** [review]
We should rewrite how we are testing the debugger to use Flutter integration tests similar to what has been done for the inspector tests. That will let us use golden image tests to verify the UI state.

**Comment by jacob314** [comment]
Fyi @grouma, @annagrin

**Comment by kenzieschmoll** [comment]
can you add a screenshot or gif to your PR description? thanks

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/tree.dart [hunk 1]
```diff
@@ -13,10 +13,27 @@ class TreeView<T extends TreeNode<T>> extends StatefulWidget {
...
+  /// has unbounded constraints in the [scrollDirection], then [shrinkWrap] must
+  /// be true.
+  ///
+  /// Shrink wrapping the content of the scroll view is significantly more
+  /// expensive than expanding to the maximum allowed size because the content
+  /// can expand and contract during scrolling, which means the size of the
+  /// scroll view needs to be recomputed whenever the scroll position changes.
+  ///
+  /// Defaults to false.
+  final bool shrinkWrap;
```
Comment (grouma) [inline packages/devtools_app/lib/src/tree.dart:35]: Is this a copied comment? It doesn't seem appropriate here?

**Review by grouma — APPROVED** [review]

**Comment by jacob314** [comment]
Here is a screenshot.
Screen Shot 2021-02-02 at 8 51 24 AM

**Comment by jacob314** [comment]
One note is the console has a mix of fonts. We could make everything monospace if we want.

File: packages/devtools_app/lib/src/tree.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/tree.dart:35]: Yeah this is the boilerplate Flutter comment on what shrinkWrap does. I have now swapped to the TLDR of when it is needed.

**Comment by jacob314** [comment]
I think we should make all expandable variables in the debugger monospace font for consistency. Tweaking that.

**Comment by jacob314** [comment]
Please review the follow up changes since the last review. I needed to make a small refactor to resolve fighting between SelectableText widgets and InkWell widgets. Also fixed some tests that were broken and made JSON dumps fully selectable.

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**